src/utils/leakdetector/leakdetector.cpp

```diff
@@ -36,16 +39,18 @@ LeakDetector::~LeakDetector() {
     }
 
     hasLeaks = true;
-    out << className << Qt::endl;
+    (*m_out) << className << Qt::endl;
 
     for (auto l = i->begin(); l != i->end(); ++l) {
-      out << "  - ptr: " << l.key() << " size:" << l.value() << Qt::endl;
+      (*m_out) << "  - ptr: " << l.key() << " size:" << l.value() << Qt::endl;
     }
   }
 
   if (!hasLeaks) {
-    out << "No leaks detected." << Qt::endl;
+    (*m_out) << "No leaks detected." << Qt::endl;
   }
+
+  (*m_out).flush();
```

Comment:
Why add this change?

Reply:
Given that we now can add any QIODevice as an output target to leakdetector `<<` does not necessarily would send it all out, that depends on the device impl. Imagine it's a tcp stream sending 512 bytes packages at a time and we only feed "<< 'a'", depending on the stream state it might buffer a bit until it can write a full package.  
If we call flush we make sure that everything is written out sync at this point if it has not been yet. 
I.e if we would ever pipe this to a file we could end up with just a few parts written and the rest lost in the buffer. :)